Switch to Material 3

7h30n3 opened the pull request:
* Rename parameter
* Remove obsolete parameter
* Streamline used theme colors
* Replace Login Button with FAB
* Rename Action Buttons for Select/Alert Dialog

Robbendebiene commented on lib/widgets/map_overlay/zoom_button.dart:
```diff
@@ -13,9 +14,10 @@ class ZoomButton extends StatelessWidget {
     return Material(
+      borderRadius: BorderRadius.circular(Default.borderRadius),
       elevation: Theme.of(context).floatingActionButtonTheme.elevation ?? 8.0,
-      shape: Theme.of(context).floatingActionButtonTheme.shape,
-      color: Theme.of(context).colorScheme.secondary,
+      color: Theme.of(context).colorScheme.primaryContainer,
```
I don't like directly using the static default theme. What was wrong with the previous solution?

Also on the one side we do no longer force the FAB border radius:
The following was removed:
```
     shape: const RoundedRectangleBorder(
          borderRadius: BorderRadius.all(
              Radius.circular(Default.borderRadius)
          )
      )
```
On the other side we apply a hardcoded value to the zoom buttons.

Robbendebiene replied:
So you prefer inconsistency (especially when a new future material update kicks in) just to always use the latest material default values?
I'm not sure if this is a good idea even without the inconsistency. The current app design is always tailored to a specific material version. So we will always have to make adjustments when a new version comes out.

Isn't the whole point of theming to define a consistent style at one place for the entire app?

7h30n3 replied:
Reverted it.

Robbendebiene reviewed (changes requested):
Map layer switcher now gets the accent color when active:
![grafik](https://user-images.githubusercontent.com/13716661/192228894-2eaa2998-d432-4953-a380-52ab2b23676f.png)

Back arrow color changed:
![grafik](https://user-images.githubusercontent.com/13716661/192229519-2b353b95-69ae-465d-9483-2735e0938c2b.png)

Header color changes when scrolling + no header shadow:
![grafik](https://user-images.githubusercontent.com/13716661/192229733-1c75b9cd-b663-4912-84dc-0380382e5490.png)

Toggle button not correctly colored:
![grafik](https://user-images.githubusercontent.com/13716661/192230094-743aeb76-0df4-4b71-951e-e49548fdd5c0.png)

Color + unpleasant paddings in my eye, very big headline. Maybe we can tweak some of these things:
![grafik](https://user-images.githubusercontent.com/13716661/192230486-2c6bc125-5039-4bab-b2a7-bb3951aa4196.png)

Very large button with a lot of elevation. I still think that FAB is **not** the right choice here.

![grafik](https://user-images.githubusercontent.com/13716661/192231236-67b311bb-8349-4145-b91b-e5597077600b.png)

Missing shadow:
![grafik](https://user-images.githubusercontent.com/13716661/192231428-61f6bd07-1544-4804-bb58-acd757f96364.png)

Color looks brownish in contrast to the previous version or the Sidebar and Sheet:
![grafik](https://user-images.githubusercontent.com/13716661/192232645-79ea39bc-c6bb-4a65-9d0d-0dd48326b45d.png)

7h30n3 commented on lib/widgets/home_sidebar.dart:
```diff
@@ -255,25 +255,11 @@ class UserAccountActionButton extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return DecoratedBox(
-      decoration: BoxDecoration(
-            borderRadius: BorderRadius.circular(12),
-            boxShadow: [
-              BoxShadow(
-                spreadRadius: 3,
-                blurRadius: 4,
-                color: Theme.of(context).colorScheme.shadow
-              )
-            ]
-          ),
-      child: FloatingActionButton.small(
+    return FloatingActionButton.small(
         heroTag: null,
-        backgroundColor: Theme.of(context).colorScheme.onPrimary,
         onPressed: onTap,
-        elevation: 0,
         child: Icon(icon),
-      ),
-    );
```
I understand. That code change must have slipped through from my testing to redesign the drawer header.

7h30n3 commented:
> Map layer switcher now gets the accent color when active: ![grafik](https://user-images.githubusercontent.com/13716661/192228894-2eaa2998-d432-4953-a380-52ab2b23676f.png)

I changed that behavior to be more in line with the selected "Location" button and the small specific map layer buttons themself.

> Back arrow color changed: ![grafik](https://user-images.githubusercontent.com/13716661/192229519-2b353b95-69ae-465d-9483-2735e0938c2b.png)
 
That's an upstream issue and doesn't persist anymore using Flutter 3.4.0

> Header color changes when scrolling + no header shadow: ![grafik](https://user-images.githubusercontent.com/13716661/192229733-1c75b9cd-b663-4912-84dc-0380382e5490.png)

The changing header color is a material 3 thing, but I don't know about the missing shadow. I will take a look there.

> Toggle button not correctly colored: ![grafik](https://user-images.githubusercontent.com/13716661/192230094-743aeb76-0df4-4b71-951e-e49548fdd5c0.png)

That's an upstream issue and doesn't persist anymore using Flutter 3.4.0

> Color + unpleasant paddings in my eye, very big headline. Maybe we can tweak some of these things: ![grafik](https://user-images.githubusercontent.com/13716661/192230486-2c6bc125-5039-4bab-b2a7-bb3951aa4196.png)

The wrong color of the toggleable is an upstream issue and doesn't persist anymore using Flutter 3.4.0
I removed some of our styling to get these dialogs more in line with material 3

> Very large button with a lot of elevation. I still think that FAB is **not** the right choice here.

I will replace it with a filled button or tonal filled button: https://m3.material.io/components/buttons/guidelines#9ecffdb3-ef29-47e7-8d5d-f78b404fcafe

> ![grafik](https://user-images.githubusercontent.com/13716661/192231236-67b311bb-8349-4145-b91b-e5597077600b.png)
> 
> Missing shadow: ![grafik](https://user-images.githubusercontent.com/13716661/192231428-61f6bd07-1544-4804-bb58-acd757f96364.png)

I will fix it.
 
> Color looks brownish in contrast to the previous version or the Sidebar and Sheet: ![grafik](https://user-images.githubusercontent.com/13716661/192232645-79ea39bc-c6bb-4a65-9d0d-0dd48326b45d.png)

I will fix it.

Robbendebiene commented:
> I changed that behavior to be more in line with the selected "Location" button and the small specific map layer buttons themself.

I could see this working if we had a secondary color. Then the main FAB would get the primary when active and the small FABs get the secondary color when active. But since we don't have this distinction I find it confusing that two nearby buttons share the same selection style but not the same state. The main button is just like a select box and therefore only holds an ephemeral state while the smaller buttons represent an actual state that gets saved across app restarts and that also affects how the app works, wherefore I would consider it more important.

> The changing header color is a material 3 thing, 

Feels unnecessary to me.

> I removed some of our styling to get these dialogs more in line with material 3

The alert dialog looks off to me compared to the rest of the apps paddings/font sizes.

> That's an upstream issue and doesn't persist anymore using Flutter 3.4.0

I would then delay the merge of this PR till 3.4. hits stable.

Robbendebiene commented on lib/widgets/home_sidebar.dart:
```diff
@@ -320,10 +328,14 @@ class LoginInfoHeader extends StatelessWidget {
+            ).copyWith(
+                elevation: ButtonStyleButton.allOrNull(0.0)),
+                onPressed: onLoginTap,
+                child: const Text('Anmelden'),
```
Can you please explain this to me. I would expect that the following does the same.

```
style: ElevatedButton.styleFrom(
  foregroundColor: Theme.of(context).colorScheme.primary,
  backgroundColor: Theme.of(context).colorScheme.onPrimary,
  elevation: 0,
)
onPressed: onLoginTap,
child: const Text('Anmelden'),
```
If it has any purpose I think a comment would be good. Otherwise we might remove this in the future as part of some simplification or re-factoring.
            
Btw the indention of `onPressed` and `child` confused me.

7h30n3 replied:
Fixed it.

Robbendebiene commented on lib/widgets/home_sidebar.dart:
```diff
@@ -20,9 +20,10 @@ class HomeSidebar extends StatefulWidget {
-      child: Drawer(
+    return Drawer(
+        width: min(MediaQuery.of(context).size.width * 0.65, 300),
+        backgroundColor: Theme.of(context).colorScheme.background,
+        shape: const Border(),
```
Maybe define this in the theme so it is applied "globally"?
